# Introduction

Note: Quality control for NGS raw data (e.g DNAseq or RNAseq) is in essence the same. 

The differences in the analyses happen after: from the selections of the mapping tools to the downstream analysis.

For any question, you can contact me: mitra [at] ebi.ac.uk 

This tutorial will illustrate how to check the quality of your raw reads with standalone tools (see [Software requirements] (https://github.com/barzine/TeachingMaterial#software-requirements). Keep in mind that this is a rapidly evolving field and that this document is not intended as a review of the many tools available to perform each step; instead, we will cover one of the many existing workflows to analyse this type of data.

We will be working with a subset of a publicly available dataset from *Drosophila melanogaster*, which is available both in the Short Read archive ([SRP001537](http://www.ebi.ac.uk/ena/data/view/SRP001537) - raw data). For more information about this dataset please refer to the original publication ([Brooks et al. 2010](http://genome.cshlp.org/content/early/2010/10/04/gr.108662.110)).


