This tutorial will illustrate how to use standalone tools, together with R and Bioconductor for the analysis of RNA-seq data. We will also use one meta-pipeline [IRAP] (https://github.com/nunofonseca/irap). Keep in mind that this is a rapidly evolving field and that this document is not intended as a review of the many tools available to perform each step; instead, we will cover *one* of the many existing workflows to analyse this type of data.

We will be working with a subset of a publicly available dataset from *Drosophila melanogaster*, which is available:
* As raw data: in the Short Read archive [SRP001537](http://www.ebi.ac.uk/ena/data/view/SRP001537) or through ArrayExpress/ENA [E-GEOD-18508] (http://www.ebi.ac.uk/arrayexpress/experiments/E-GEOD-18508/)
* and also as processed data: in Bioconductor as a package: [pasilla](http://www.bioconductor.org/packages/release/data/experiment/html/pasilla.html)

For more information about this dataset please refer to the original publication [(Brooks et al. (2010)] (http://genome.cshlp.org/content/early/2010/10/04/gr.108662.110).

The tools and R packages that we will be using during the practical are listed below (see [Software requirements](https://github.com/Functional-Genomics/TeachingMaterial#software-requirements)) and the necessary data files can be found [here](http://www.ebi.ac.uk/~mitra/courses/EMBLPredocCourse2015/RNASeq.tar.gz). After dowloading and uncompressing the `tar.gz` file, you should have the following directory structure in your computer:

```
DATA                                  # data used for the practicals
|-- demultiplexing                    # multiplexed data !!! Not used in the project - FYI only
|-- eqtl                              # data used in the eqtl practical !!! Not used in the project - FYI only
|-- fastq                             # fastq files -> starting point
|-- mapped                            # mapped data: BAM files
|-- QCreports                         # precomputed QC report
`-- IRAP_example                          # Directory setup for IRAP (raw_data +reference) + its output
    |-- data
    |   |-- contamination                 # E.coli reference
    |   |-- raw_data                      # fastq files
    |   |-- reference
    |   |   |--drosophila_melanogaster    # All the reference files are in this directory
    `-- E-GEOD-18508                      # output of IRAP
        | ...
```



