This tutorial will illustrate how to use standalone tools, together with R and Bioconductor for the analysis of RNA-seq data. We will also use one meta-pipeline [IRAP] (https://github.com/nunofonseca/irap). Keep in mind that this is a rapidly evolving field and that this document is not intended as a review of the many tools available to perform each step; instead, we will cover *one* of the many existing workflows to analyse this type of data.

We will be working with a subset of a publicly available dataset from *Homo sapiens*, which is available:
* As raw data:
    * in the Short Read archive [ERP003613](http://www.ebi.ac.uk/ena/data/view/ERP003613) and 
    * in ArrayExpress [E-MTAB-2836] (http://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-2836/) 
* but also as processed data 
    *  in the authors' website [Protein Atlas] (http://www.proteinatlas.org/humanproteome/tissue+specific) and 
    * [EBI Gene Expression Atlas] (http://www.ebi.ac.uk/gxa/experiments/E-MTAB-2836). 

For more information about this dataset please refer to the original publication ([Uhlen et al. Science (2015) - Tissue-based map of the human proteome. DOI: 10.1126/science.1260419] (http://dx.doi.org/10.1126/science.1260419))..  

The tools and R packages that we will be using during the practical are listed below (see [Software requirements](https://github.com/Functional-Genomics/TeachingMaterial#software-requirements)) and the necessary data files can be found [here](http://www.ebi.ac.uk/~mitra/courses/CG15/RNASeq.tar.gz). After dowloading and uncompressing the `tar.gz` file, you should have the following directory structure in your computer:

```
RNAseq
|-- reference               # reference info (e.g. genome sequence and annotation)
`-- data
    |-- raw                 # raw data: fastq files
    |-- demultiplexing      # multiplexed data !not used for this course
    |-- mapped              # mapped data: BAM files
    `-- RData               # R environment for each part with the different generated object
```
