## Filtering FASTQ files
After analysing the QA report, one might want to discard some of the reads based on several criteria, such as quality and nucleotide composition. We will use two different tools to perform these filtering steps: [PRINSEQ](http://prinseq.sourceforge.net/) and [fastq-mcf](https://code.google.com/p/ea-utils/).

PRINSEQ offers a wide range of options for filtering and we can learn about them in the manual:

```bash
prinseq-lite -h
```

Based on what we have learned from the QA report, we could decide to apply the following filters:

```bash
cat ERR315494_1.fastq | prinseq-lite \
    -fastq stdin \
    -out_good ERR315494_1_filt1 \
    -out_bad null \
    -trim_qual_right 30 -min_len 32 \
    -ns_max_p 5 \
    -lc_method dust -lc_threshold 10
    
cat ERR315494_2.fastq | prinseq-lite \
    -fastq stdin \
    -out_good ERR315494_2_filt1 \
    -out_bad null \
    -trim_qual_right 30 -min_len 32 \
    -ns_max_p 5 \
    -lc_method dust -lc_threshold 10
```

### Exercise1:  
Which are the criteria that we are using to discard reads? 
[Hint](https://github.com/Functional-Genomics/TeachingMaterial/blob/Cancer-Genomics-07-2015/solutions/_filtering_fastq_ex1.md)

### Exercise 2: 
Which extra option should we have had to use if our files had been in phred 64 format?
[Solution](https://github.com/Functional-Genomics/TeachingMaterial/blob/Cancer-Genomics-07-2015/solutions/_filtering_fastq_ex2.md)

### Exercise 3: 
After filtering the fastq file it is not a bad idea to obtain a QA report again to decide whether we are happy with the results. Once again, the QC reports are already provided in the `QCreports/filt1` folder.  
Do you think we got rid of the main issues spotted initially?
[Solution](https://github.com/Functional-Genomics/TeachingMaterial/blob/Cancer-Genomics-07-2015/solutions/_filtering_fastq_ex3.md)

### Exercise 4 :
Usually it is also useful to keep track of the number of reads available in the fastq file both before and after the filtering step. Can you gather this information from the FastQC reports? Given that we are working with paired-end data, do you see any limitation?
Now imagine we were working with a bigger number of files; can you come up with a more automated way to check that?
[Solution](https://github.com/Functional-Genomics/TeachingMaterial/blob/Cancer-Genomics-07-2015/solutions/_filtering_fastq_ex4.md)

The filtering step might become complicated if you are working with paired-end data, since you have to be sure that both fastq files (one for each read pair) contain the same number of reads in the same order. There are some tools available that simplify this task, for example fastq-mcf. We can print a list of the avaiable options just by typing in the name of the tool:
```bash
fastq-mcf
```

We observe that a main functionality of fastq-mcf is to remove adapters from the fastq file. For this reason we need to provide a fasta file with the adapter sequences. When you design your experiment, you know before hand the sequences of the adaptors. These sequences can also be found online (e.g. some [Illumina adaptors](http://supportres.illumina.com/documents/documentation/chemistry_documentation/experiment-design/illumina-customer-sequence-letter.pdf)). In our case, we have decided to check only for the standard Illumina paired-end adapter:
```bash
cat adapters.fa
```

Now that we have a good understanding of the required input we can proceed to execute the tool, trying to match the options that we used previously with PRINSEQ:
```bash
fastq-mcf adapters.fa ERR315494_1_filt1.fastq ERR315494_1_filt2.fastq \
    -o ERR315494_1_filt2.fastq -o ERR315494_2_filt2.fastq \
    -q 30 -P 33 -l 32 --max-ns 1
```

### Exercise 5:
How does the QA report look like this time? Do we have the same number of reads in each file?
[Solution](https://github.com/Functional-Genomics/TeachingMaterial/blob/Cancer-Genomics-07-2015/solutions/_filtering_fastq_ex5.md)

### Exercise 6:
Something else that one might want to check is the read length. How long were our reads before we started with the filtering step? How long are they now?
[Solution](https://github.com/Functional-Genomics/TeachingMaterial/blob/Cancer-Genomics-07-2015/solutions/_filtering_fastq_ex6.md)

Depending on the filtering options used, we might end up with a set of reads with different lengths. *A priori*, this should not be a limitation, but we might encounter some difficulties in the downstream analyses, depending on the tools we want to use. For this reason, if we have a clear idea of what we want to do with the data, it is always a good practice to check the requirements for the tools that we are planning to use before taking any steps. If that is not the case, in order to be on the safe side, we can use filtering options that affect all reads equally, eg:

```bash
cat ERR315494_pe_1_filt2.fastq | prinseq-lite \
    -fastq stdin \
    -out_good ERR315494_pe_1_filt3 \
    -out_bad null \
    -trim_right 5
    
cat ERR315494_pe_2_filt2.fastq | prinseq-lite \
    -fastq stdin \
    -out_good ERR315494_pe_2_filt3 \
    -out_bad null \
    -trim_right 5
```

### Exercise 7:
Let us generate the QA reports one last time. How do they compare to the previous ones?
[Solution](https://github.com/Functional-Genomics/TeachingMaterial/blob/Cancer-Genomics-07-2015/solutions/_filtering_fastq_ex7.md)

In conclusion, there are many filtering combinations that you can apply, and the specific options will largely depend on the type of data and the posterior analyses. We recommend to check the [PRINSEQ manual](http://prinseq.sourceforge.net/manual.html) for a nice overview on the topic.

