## Normalising counts
### With RPKMs
While in the previous sections the data was derived from a single sample, in this exercise we will work with the precomputed counts for all the samples in our experiment. For this we are going to use a matrix of raw counts (mapped with tophat2 and quantified with HTS-seq count) provided by IRAP:
`../IRAP_example/E-MTAB-2836/tophat2/htseq2/genes.raw.htseq2.tsv`

```rconsole
rawcounts<-read.table('../IRAP_example/E-MTAB-2836/tophat2/htseq2/genes.raw.htseq2.tsv',
header=TRUE, rownames=1, sep='\t')
````
We will also need an annotation matrix providing the meta information of the experiment:

```rconsole
expAnnot<-data.frame(tissue=c('kidney','liver','liver','kidney','liver','kidney'),
		type=c('pe','pe','pe','pe','pe','pe'),
		sex=c('m','m','m','f','f','f'))
row.names(expAnnot)<-c("lERR315383", "lERR315394", 
			"lERR315414", "lERR315443", 
			"lERR315451", "lERR315494")
```

Now, we can create a *DESeqDataSet*, which is a data container used by DESeq2 and will be later on used to calculate the differential expressed genes 

```rconsole
library(DESeq) ## we need to access some functions directly
library(DESeq2) ## mainly a wrapper around DESeq with some nice additions
dds <- DESeqDataSetFromMatrix(countData = rawcounts,
                              colData = expAnnot,
                              design = ~ tissue)
                              
colData(dds)$tissue=factor(colData(dds)$tissue,
		levels=c("kidney","liver"))

counts=counts(dds) 
summary(counts) 
```

What is the difference between the `counts` object and the `rawcounts`? _Hint_: use the `str` or `class` functions. Have also a look at `all()`.

A common way to normalise reads is to convert them to RPKMs (or FPKMs in the case of paired-end data). This implies normalising the read counts depending on the feature size (exon, transcript, gene model...) and on the total number of reads sequenced for that library:

![RPKM formula](../img/rpkms.png)

#### Exercise 1:
Let us obtain RPKMs for the table `counts` following these steps:

* Calculate the length of the exons in the `annot` object and store the result in a vector, with the name of each element set to the corresponding gene.
  *Hint:* check the *width* and *elementMetadata* accessors.

* Obtain gene lengths by adding up the lengths of all the exons in each gene.
  *Hint:* check the functions *split* and *sapply*.

* Normalise the counts by the library size.
  *Hint:* check the function *colSums*.

* Continue normalising by gene length, but be aware that the object that contains the gene lengths and the one that contains the normalised counts might have a different number of genes.

* Finally, obtain RPKMs by multiplying by a factor of 10^9.

[Solution](https://github.com/Functional-Genomics/TeachingMaterial/blob/Cancer-Genomics-07-2015/solutions/_normalising_ex1.md)

Such a count normalisation is suited for visualisation, but sub-optimal for further analyses. A better way of normalising the data is to use either the *edgeR* or *DESeq2* packages.

### With DESeq2
RPKM normalisation is not the most adequate for certain types of downstream analysis (e.g. differential gene expression), given that it is susceptible to library composition biases. There are many other normalisation methods that should be considered with that goal in mind (see [Dillies et al. 2012](http://bib.oxfordjournals.org/content/early/2012/09/15/bib.bbs046.long) for a comparison). In this section we are going to explore the one offered within the *DESeq2* package:

```rconsole
library("Biobase") 
library("DESeq2") #if not done already before

# load the count data and the experimental design and then
# create an object of class DESeqDataSet, which is the data container used by DESeq2
# you can skip this if you have already loaded the data in the previous section

rawcounts<-read.table('../IRAP_example/E-MTAB-2836/tophat2/htseq2/genes.raw.htseq2.tsv',
header=TRUE, rownames=1, sep='\t')
expAnnot<-data.frame(tissue=c('kidney','liver','liver','kidney','liver','kidney'),
		type=c('pe','pe','pe','pe','pe','pe'),
		sex=c('m','m','m','f','f','f'))
row.names(expAnnot)<-c("lERR315383", "lERR315394", 
			"lERR315414", "lERR315443", 
			"lERR315451", "lERR315494")

dds <- DESeqDataSetFromMatrix(countData = rawcounts,
                              colData = expAnnot,
                              design = ~ tissue)
                              
colData(dds)$tissue=factor(colData(dds)$tissue,
		levels=c("kidney","liver"))

dds

# the DESeqDataSet class is a container for the information we just provided
head(counts(dds))
colData(dds)
design(dds)
```

In order to normalise the raw counts we will start by determining the relative library sizes, or *size factors* for each library. For example, if the counts of the expressed genes in one sample are, on average, twice as high as in another, the size factor for the first sample should be twice as large as the one for the other sample. These size factors can be obtained with the function *estimateSizeFactors*:

```rconsole
dds=estimateSizeFactors(dds)
sizeFactors(dds)
```

Once we have this information, the normalised data is obtained by dividing each column of the count table by the corresponding size factor. We can perform this calculation by calling the function counts with a specific argument as follows:

```rconsole
deseq_ncounts=counts(dds, normalized=TRUE)
```

**Exercise:** We have now accumulated three different versions of the same dataset: the raw counts (`counts`), the RPKM quantifications (`rpkm`) and the DESeq normalised counts (`deseq_ncounts`). How would you visualise the performance of each normalisation method in getting rid of the variation that does not associate to the experimental conditions that are being evaluated?
[Solution](https://github.com/Functional-Genomics/TeachingMaterial/blob/Cancer-Genomics-07-2015/solutions/_normalising_ex2.md)

