#Mapping eQTLs with MatrixEQTL

##Loading the MatrixEQTL library

```rconsole
library(MatrixEQTL)
```

##Location of the data 
```rconsole
#Set base dir
base.dir = "http://www.ebi.ac.uk/~ccalabre/training"

# Genotype file name
SNP_file_name = paste(base.dir, "/geuvadis_geno.txt", sep="");

# Gene expression file name
expression_file_name = paste(base.dir, "/geuvadis_pheno.txt", sep="");

# Covariates file name
covariates_file_name = paste(base.dir, "/covariates.txt", sep="");

# Output file name
output_file_name = tempfile();
```
##Set the parameters you want to use and the model

```rconsole
# Only associations significant at this level will be saved
pvOutputThreshold = 1e-2;

#The error covariance matrix. Use numeric() for homoskedastic independent errors
errorCovariance = numeric();

# Linear model to use 
useModel = modelLINEAR; #this condisers only additive effect of the genotype 
```

##Loading the data
```rconsole
#genotype
snps = SlicedData$new();
snps$fileDelimiter = "\t";      # the TAB character
snps$fileOmitCharacters = "NA"; # denote missing values;
snps$fileSkipRows = 1;          # one row of column labels
snps$fileSkipColumns = 1;       # one column of row labels
snps$fileSliceSize = 2000;      # read file in slices of 2,000 rows
snps$LoadFile(SNP_file_name);

#phenotype
gene = SlicedData$new();
gene$fileDelimiter = "\t";      # the TAB character
gene$fileOmitCharacters = "NA"; # denote missing values;
gene$fileSkipRows = 1;          # one row of column labels
gene$fileSkipColumns = 1;       # one column of row labels
gene$fileSliceSize = 2000;      # read file in slices of 2,000 rows
gene$LoadFile(expression_file_name);

#covariates
#Set to only cvrt = SlicedData$new() for no covariates
cvrt = SlicedData$new(); 
cvrt$fileDelimiter = "\t";      # the TAB character
cvrt$fileOmitCharacters = "NA"; # denote missing values;
cvrt$fileSkipRows = 1;          # one row of column labels
cvrt$fileSkipColumns = 1;       # one column of row labels
if(length(covariates_file_name)>0) {
cvrt$LoadFile(covariates_file_name);
}

```
##Running the eQTL analysis by testing any SNP-gene pairwaise association

```rconsole
me = Matrix_eQTL_engine(
snps = snps,
gene = gene,
cvrt = cvrt,
output_file_name = output_file_name,
pvOutputThreshold = pvOutputThreshold,
useModel = useModel, 
errorCovariance = errorCovariance, 
verbose = TRUE,
pvalue.hist = TRUE,
min.pv.by.genesnp = FALSE,
noFDRsaveMemory = FALSE);

unlink(output_file_name);
```
##Getting results

```rconsole
cat('Analysis done in: ', me$time.in.sec, ' seconds', '\n');
cat('Detected eQTLs:', '\n');
eqtls = me$all$eqtls
str(eqtls)
## Plot the histogram of all p-values
plot(me)
```
##Running cis and trans eQTLs separately
```rconsole
# Linear model to use
useModel = modelLINEAR; 

#loading of annotations for cis-eQTL analysis
snps_location_file_name = paste(base.dir, "/geno_annotations.txt", sep="");
gene_location_file_name = paste(base.dir, "/pheno_annotations.txt", sep="");

# Output file name
output_file_name_cis = tempfile();
output_file_name_tra = tempfile();

# Setting parameters:
##Only associations significant at this level will be saved
pvOutputThreshold_cis = 2e-2;
pvOutputThreshold_tra = 1e-2;

## Error covariance matrix
## Set to numeric() for identity.
errorCovariance = numeric();

## Distance for local gene-SNP pairs (cis-eQTL)
cisDist = 1e6;

## Load genotype data
snps = SlicedData$new();
snps$fileDelimiter = "\t";      # the TAB character
snps$fileOmitCharacters = "NA"; # denote missing values;
snps$fileSkipRows = 1;          # one row of column labels
snps$fileSkipColumns = 1;       # one column of row labels
snps$fileSliceSize = 2000;      # read file in slices of 2,000 rows
snps$LoadFile(SNP_file_name);

## Load gene expression data
gene = SlicedData$new();
gene$fileDelimiter = "\t";      # the TAB character
gene$fileOmitCharacters = "NA"; # denote missing values;
gene$fileSkipRows = 1;          # one row of column labels
gene$fileSkipColumns = 1;       # one column of row labels
gene$fileSliceSize = 2000;      # read file in slices of 2,000 rows
gene$LoadFile(expression_file_name);

## Load covariates
#Set to only cvrt = SlicedData$new() for no covariates
cvrt = SlicedData$new();
cvrt$fileDelimiter = "\t";      # the TAB character
cvrt$fileOmitCharacters = "NA"; # denote missing values;
cvrt$fileSkipRows = 1;          # one row of column labels
cvrt$fileSkipColumns = 1;       # one column of row labels
if(length(covariates_file_name)>0) {
cvrt$LoadFile(covariates_file_name);
}

## Run the analysis
snpspos = read.table(snps_location_file_name, header = TRUE, stringsAsFactors = FALSE);
genepos = read.table(gene_location_file_name, header = TRUE, stringsAsFactors = FALSE);

## to run only cis-eQTL
pvOutputThreshold_tra = 0 for cis eQTL
#to run only trans-eQTL
pvOutputThreshold_cis = 0 for trans eQTL 

#Running the eQTL analysis 
me = Matrix_eQTL_main(
snps = snps, 
gene = gene, 
cvrt = cvrt,
output_file_name = output_file_name_tra,
pvOutputThreshold = pvOutputThreshold_tra,
useModel = useModel, 
errorCovariance = errorCovariance, 
verbose = TRUE, 
output_file_name.cis = output_file_name_cis,
pvOutputThreshold.cis = pvOutputThreshold_cis,
snpspos = snpspos, 
genepos = genepos,
cisDist = cisDist,
pvalue.hist = "qqplot",
min.pv.by.genesnp = FALSE,
noFDRsaveMemory = FALSE);

unlink(output_file_name_tra);
unlink(output_file_name_cis);

## Results:

cat('Analysis done in: ', me$time.in.sec, ' seconds', '\n');
cat('Detected local eQTLs:', '\n');
cis_eqtls = me$cis$eqtls
str(cis_eqtls)
cat('Detected distant eQTLs:', '\n');
trans_eqtls = me$trans$eqtls
str(trans_eqtls)
## Plot the Q-Q plot of local and distant p-values

plot(me)
```

##Visualising results: Manhattan plots and Q-Q plots

```rconsole
#Preparing the data to visualise cis-eQTLs and load the qqman library

cis_eqtls<-me$cis$eqtls

install.packages("qqman")
library(qqman)

## making the data frame
SNP = cis_eqtls$snps
CHR = rep(22,length(SNP))
P  = cis_eqtls$pvalue
BP = snpspos$pos[match(SNP,snpspos$snp)] 
gwasResults = data.frame(SNP,CHR,BP,P)
```

## Plotting the Manhattan plot 
```rconsole
manhattan(gwasResults)
```
##Plotting the Q-Q plot
```rconsole
qq(gwasResults$P, main = "Q-Q plot of p-values")
```
