# Extra information

## Software requirements
*Note: depending on the topics covered in the course some of these tools might not be used.*

* Standalone tools:
    * [FastQC](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/)
    * [PRINSEQ](http://prinseq.sourceforge.net/)
    * [eautils](https://code.google.com/p/ea-utils/)
    * [samtools](http://sourceforge.net/projects/samtools/)
    * [IGV](http://www.broadinstitute.org/software/igv/download)
    * [htseq-count](http://www-huber.embl.de/users/anders/HTSeq/doc/count.html)
    * [docker](https://www.docker.com/)

* Bioconductor packages:
    * [GenomicRanges](http://www.bioconductor.org/packages/release/bioc/html/GenomicRanges.html)
    * [GenomicAlignments](http://bioconductor.org/packages/release/bioc/html/GenomicAlignments.html)
    * [Rsamtools](http://www.bioconductor.org/packages/release/bioc/html/Rsamtools.html)
    * [biomaRt](http://www.bioconductor.org/packages/release/bioc/html/biomaRt.html)
    * [pasilla](http://www.bioconductor.org/packages/release/data/experiment/html/pasilla.html)
    * [DESeq2](http://www.bioconductor.org/packages/release/bioc/html/DESeq2.html)
    * [DEXSeq](http://www.bioconductor.org/packages/release/bioc/html/DEXSeq.html)


## Other resources

### Course data
* [Complete course data, including command outputs and R sessions](http://www.ebi.ac.uk/~mitra/courses/RNASeq.tar.gz)

### Tutorials
* [Course materials available at the Bioconductor website](http://www.bioconductor.org/help/course-materials/)
* [Online training resources at the EBI website](http://www.ebi.ac.uk/training/online/course-list?topic%5B%5D=13&views_exposed_form_focused_field=)
* [R and Bioconductor tutorial by Thomas Girke](http://manuals.bioinformatics.ucr.edu/home/R_BioCondManual)
* Do not forget to check the documentation for the packages used in the practical!

### Cheat sheets
* [R reference card](http://cran.r-project.org/doc/contrib/Short-refcard.pdf)
* [Unix comand line cheat sheet](http://sites.tufts.edu/cbi/files/2013/01/linux_cheat_sheet.pdf)


# Aknowledgments
This tutorial has been inspired on material developed by Mar Gonzalez-Porta, Liliana Greger, Ângela Gonçalves, Nicolas Delhomme, Simon Anders and Martin Morgan, who we would like to thank and acknowledge. Many thanks to Claudia Calabrese & Fatemeh Zamanzad for having participate in the elaboration of this course. Special thanks must go to Mar Gonzalez-Porta, Claudia Calabrese & Fatemeh Zamanzad with whom we have been teaching.

